common/fs/src/cache/mod.rs

```diff
@@ -310,11 +338,13 @@ where
             }
         };
 
-        let entry = unsafe { entry_ptr.as_ref() };
-        let mut path = self.resolve_direct_path(entry);
-        path.push(name);
-
-        self.remove(&path, events);
+        if let Some(entry) = _entries.get(entry_ptr) {
+            let mut path = self.resolve_direct_path(&entry.borrow(), _entries);
+            path.push(name);
+            self.remove(&path, events, _entries);
+        } else {
+            error!("Failed to find entry");
```

Comment:
Better than crashing :)

Reply:
Yeh, I havn't worked out if this should be an error that's handled somwhere higher up. I'm pretty sure it's not a big deal as this is a cache, so the worst case is we end up repopulating something.